wallet/udb/addressmanager.go

```diff
@@ -2315,6 +2315,34 @@ func (m *Manager) PrivateKey(ns walletdb.ReadBucket, addr stdaddr.Address) (key
 	return key, key.Zero, nil
 }
 
+// HavePrivateKey returns whether the private key for a P2PK or P2PKH address is
+// available when the wallet or account is unlocked.
+func (m *Manager) HavePrivateKey(ns walletdb.ReadBucket, addr stdaddr.Address) (bool, error) {
+	defer m.mtx.RUnlock()
+	m.mtx.RLock()
+
+	if m.watchingOnly {
+		return false, nil
+	}
+
+	id, err := addressID(addr)
+	if err != nil {
+		return false, nil
+	}
```

Comment:
Better to return err here because we were expecting P2PK or P2PKH addr which should have an ID? ofc not sure tho.

Reply:
`addressID`uses a check for `Hash160er` which also supports P2SH, Schnorr, etc. This will really only err for a completely unknown address type, in which case we don`t have the private key anyway, so this fulfills the API for this func.